docker/wizard.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+set -e
+
+function confirm() {
+    echo
+    read -p "$1 (y/n) "
+    if [[ $REPLY =~ ^[Yy]$ ]]; then
+        return 0
+    fi
+    if [[ $REPLY =~ ^[Nn]$ ]]; then
+        return 1
+    fi
+    echo "Answer must be one of y/n."
+    confirm "$1"
+}
+
+function assert_file_present() {
+    if ! [ -f $1 ]; then
+        echo "Missing file: $1. Aborting."
+        exit 1
+    fi 
+}
+
+cat - <<EOF
+*** Keywhiz Setup Wizard ***
+
+The Keywhiz Setup Wizard will take you through a step-by-step process to
+bootstrap a Keywhiz server container. This includes generating a new content
+encryption key, adding an initial admin user, as well as adding certificates
+for the Keywhiz server to run. All generated keys and secrets will be stored in
+the /secrets directory inside the container, and an empty H2 database will be
+created in /data. You should make sure you have persistent volumes mounted in
+those places!
+
+Note that if you wish to run Keywhiz with an external database you can do so
+by supplying your own config file via the KEYWHIZ_CONFIG environment variable.
+The wizard will set up an H2 database by default, which should be sufficent
+for most deployments (unless you expect a very high volume of traffic).
+EOF
+
+if ! confirm "The wizard will overwrite previously generated data if present. Proceed?"; then
+    exit 1
+fi
+
+echo
+echo -n "Generating a new cookie secret... "
+export COOKIE_KEY_PATH="/secrets/cookie.key.base64"
+head -c32 /dev/urandom | base64 > $COOKIE_KEY_PATH
+echo "done"
+
+echo -n "Generating a new content encryption key... "
+export CONTENT_KEYSTORE_PASSWORD=`head -c16 /dev/urandom | xxd -p`
+export CONTENT_KEYSTORE_PATH="/secrets/content-encryption-key.jceks"
+rm -f $CONTENT_KEYSTORE_PATH
+keytool \
+    -genseckey -alias basekey -keyalg AES -keysize 128 -storepass "$CONTENT_KEYSTORE_PASSWORD" \
+    -keypass "$CONTENT_KEYSTORE_PASSWORD" -storetype jceks -keystore $CONTENT_KEYSTORE_PATH
+echo "done"
+
+echo
+echo "Keywhiz requires a SSL/TLS server certificate and private key to run."
+echo "You can import your own cert, or have the wizard generate one."
```

Comment:
Oh, whoops, no; I dropped that functionality. It seemed weird to generate it in the container and then dump the private key (which would be necessary to to make client certs). I'd rather just have people run that on their own.